storm-client/src/jvm/org/apache/storm/windowing/TimeTriggerPolicy.java

```diff
@@ -62,7 +63,9 @@ public void reset() {
 
     @Override
     public void start() {
-        executorFuture = executor.scheduleAtFixedRate(newTriggerTask(), duration, duration, TimeUnit.MILLISECONDS);
+        // initial delay is slightly less than the duration so that the initial tuples wont't expire due to time drift
+        long initialDelay = duration - Math.min((long) (duration * .1), 100);
+        executorFuture = executor.scheduleAtFixedRate(newTriggerTask(), initialDelay, duration, TimeUnit.MILLISECONDS);
```

Comment:
Subsequent scheduling will happen after "duration" which would be the window sliding interval.